lib/Parse/Lexer.cpp

```diff
@@ -1189,6 +1189,27 @@ unsigned Lexer::lexUnicodeEscape(const char *&CurPtr, Lexer *Diags) {
   return CharValue;
 }
 
+/// isElidedNewLineEscape - Check for valid elided newline escape and
+/// move pointer passed in to the character after the end of the line.
+static bool isElidedNewLineEscape(const char *&CurPtr, ssize_t Offset) {
+  const char *TmpPtr = CurPtr + Offset;
+  while (*TmpPtr) {
+    switch (*TmpPtr++) {
+    case ' ': case '\t':
+      continue;
+    case '\r':
+      if (*TmpPtr == '\n')
+        TmpPtr++;
+      LLVM_FALLTHROUGH;
+    case '\n':
+      CurPtr = TmpPtr;
+      return true;
+    default:
+      break;
+    }
+  }
+  return false;
+}
```

Comment:
I realize much of the Lexer is inconsistent in this usage, but Clang's CharInfo.h (which is used in this file) does provide `isWhitespace`, `isHorizontalWhitespace` and `isVerticalWhitespace`. 

@jrose-apple or @nkcsgexi or anyone who would know better, is it desirable to use these instead?

Similarly for the rest of this patch